src/odemis/gui/comp/viewport.py

```diff
@@ -324,6 +324,11 @@ def setView(self, view, tab_data):
         if model.hasVA(tab_data, "zPos"):
             tab_data.zPos.subscribe(self._on_zPos_change, init=True)
 
+        # Listen to the stage position, if the view has the stage and the stage position TextCtrl
+        # has been set to be shown (by default it is hidden)
+        if hasattr(view, "stage_pos") and self.bottom_legend.stage_pos_text.IsShown():
+            view.stage_pos.subscribe(self._on_stage_pos_change, init=True)
+
```

Comment:
That feels reversed organization. Now the child class needs to make sure the stage_pos_text is shown, before calling the parent setView. Moreover the idea of `set_stage_pos_label()` is that it takes care of hidding/showing the internal widgets.
I'd suggest to entirely remove the code from this class. Then, in FastEM*Viewport classes, override `.setView()` to add:
```python
if hasattr(view, "stage_pos"):
        view.stage_pos.subscribe(self._on_stage_pos_change, init=True)
```
(the `if` might not be necessary, as normally there is always a stage on the FASTEM)

Reply:
Ahh, I misunderstood your previous comment